lib/WebpackError.js

```diff
@@ -4,8 +4,81 @@
 */
 "use strict";
 
+const createHash = require("./util/createHash");
+
+const hash = Symbol();
+const meta = Symbol();
+
 module.exports = class WebpackError extends Error {
+	constructor(
+		message,
+		line = 0,
+		column = 0,
+		columnEnd = 0,
+		category = "error"
+	) {
+		// TODO: enable message assertion during construction in a commit to follow
+		// if (!message) {
+		// 	WebpackError.deprecate("message must be a non-null, non-empty value.");
+		// }
+		// TODO: enable message length limit at a commit to follow
+		// else if (message.length > 160) {
+		// 	WebpackError.deprecate("message cannot be longer than 80 characters.");
+		// }
+
+		super(message);
+
+		this[hash] = createHash("md4");
+		this[meta] = {};
```

Comment:
`Object.create(null)` or `{__proto__: null}` would be better

Reply:
Do you have documentation I can review showing why those two alternatives are better than `{}`?

Reply:
No I don't have one. The purpose of these 2 alternatives is to have an object with a `null` prototype. Lookups on it would be faster since it doesn't require traversing the proto chain.

Reply:
OK. Small gains are small gains. I'll update that.